index.src.html

```diff
@@ -830,18 +856,29 @@ <h4 id="algorithm-create" algorithm>Create a `Credential`</h4>
             types in order to support a "sign-up" use case. For the moment, though, we're punting
             on that by restricting the dictionary to a single entry.
 
-    5.  Let |p| be [=a new promise=].
+    6.  Let |p| be [=a new promise=].
 
-    6.  Run the following steps [=in parallel=]:
+    7.  Let |origin| be |settings|'s [=environment settings object/origin=].
 
-        1.  Let |r| be the result of executing |interfaces|[0] {{Credential/[[Create]](options)}}
-            internal method on |options|.
+    8.  Run the following substeps [=in parallel=]:
 
-        2.  If |r| is an [=exception=], [=reject=] |p| with |r|.
+        1.  Let |r| be the result of executing |interfaces|[0]'s {{Credential/[[Create]](origin, options)}}
+            internal method given |origin| and |options|.
 
-            Otherwise, [=resolve=] |p| with |r|.
+        2.  If |r| is an [=exception=], [=reject=] |p| with |r|, and terminate these substeps.
 
-    7.  Return |p|.
+        3.  If |r| is a {{Credential}}, [=resolve=] |p| with |r|, and terminate these substeps.
+
+        4.  Assert: |r| is a <code>[=constructCredential=]</code> [=struct=].
+
+        5.  Let |result| be the result of [=queue a task|queuing a task=] to run
```

Comment:
Yeah, as @mikewest said, queuing a task doesn't yield a result. Instead, I'd write:

5. [=Queue a task=] on |global|'s [=DOM manipulation task source=] to run:
   1. [=Resolve=] |p| with the result of [=promise-calling=] `<code>|r|.[=constructCredentialCallback=]()</code>`.

Mike's suggestion of roughly

5. [=Queue a task=] on |global|'s [=DOM manipulation task source=] to run `<code>|r|.[=constructCredentialCallback=]</code>` passing *p*.

Would also work (handwaving the fact that we're actually passing *p*'s [stored resolve and reject functions](https://www.w3.org/2001/tag/doc/promises-guide/#shorthand-manipulating)), but it's maybe a little less clear that this function is just supposed to resolve a `Credential` into *p*.

Reply:
I think Jeffrey's language is probably more precise, but I'd be happy with either of these formulations.

Reply:
Ok, thx.  since |r| at this point represents an algorithm, am not sure what notation to use, so am giving it a shot leveraging Jeffrey's language and u guys can pick at it :)